File: guides/micronaut-token-propagation/userecho/java/src/main/java/example/micronaut/UserController.java
package example.micronaut;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;

import java.security.Principal;

@Controller("/user") // <1>
public class UserController {

    @Secured(SecurityRule.IS_AUTHENTICATED) // <2>
    @Produces(MediaType.TEXT_PLAIN) // <3>
    @Get // <4>
    String index(Principal principal) { // <5>
        return principal.getName();
    }
}

File: guides/micronaut-token-propagation/userecho/python/src/example/micronaut/user_controller.py
from java.security import Principal
from micronaut.http import MediaType
from micronaut.http.annotation import Get, Produces
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule


@Secured(SecurityRule.IS_AUTHENTICATED)
@Produces(MediaType.TEXT_PLAIN)
@Get("/user")
def index(principal: Principal) -> str:
    return principal.getName()
